caffe2/operators/layer_norm_op.cc

```diff
@@ -186,16 +186,21 @@ to the end.)
 
 // Register layer norm with c10
 namespace {
+struct State final : public c10::KernelState {
+    at::optional<at::Tensor> scale = at::nullopt;
+    at::optional<at::Tensor> bias = at::nullopt;
+};
+
 template <class DataType>
-c10::IValue layer_norm_c10(c10::ArrayRef<c10::IValue> inputs, c10::KernelState* state) {
+c10::IValue layer_norm_c10(c10::ArrayRef<c10::IValue> inputs, c10::KernelState* state) { // TODO Pass in correct state type
```

Comment:
what prevents us from passing the right type now?

Reply:
That it'd need a wrapper function doing the casting work. I'll have to do perf experiments if that wrapper function on each call is an issue or not.